lesson_28/tests/test_homework28.py

```diff
@@ -0,0 +1,180 @@
+"""Hometask 28."""
+
+import random
+
+import pytest
+from locators import (HOME_PAGE_LOCATORS, SIGN_IN_ERROR_LOCATORS,
+                      SIGN_IN_LOCATORS, SIGN_IN_SUCCESS_LOCATORS,
+                      SIGN_UP_ERROR_LOCATORS, SIGN_UP_LOCATORS)
+from pages.pom_garage import GaragePage
+from pages.pom_homepage import HomePage
+from pages.pom_sign_in import SignIn
+from pages.pom_sign_up import SignUp
+from selenium.webdriver.support import expected_conditions as EC
+from selenium.webdriver.support.wait import WebDriverWait
+
+# Configuration
+BASE_URL = 'https://guest:welcome2qauto@qauto2.forstudy.space'
+ELEMENT_WAIT_TIMEOUT = 5
+
+# Test data
+NAME = 'Adam'
+LAST_NAME = 'Smith'
+PASSWORD = 'Test123123'
+BAD_PASSWORD = '123'
+MISMATCH_PASSWORD = 'Test321321'
+EMAIL = f'Test+{random.randint(1000, 5555)}@google.com'
+
+
+def test_login_with_guest_acc(driver):
+    """Test for logging in and out as a guest."""
+    home_page = HomePage(driver)
+    garage_page = GaragePage(driver)
+    home_page.login_as_guest(ELEMENT_WAIT_TIMEOUT)
+    home_page.logout_via_menu(ELEMENT_WAIT_TIMEOUT, garage_page)
+    home_page.logout_via_dropdown(ELEMENT_WAIT_TIMEOUT, garage_page)
+
+
+@pytest.mark.parametrize(
+    'name, last_name, email, password, re_password, expected_message',
+    [
+        ('', LAST_NAME, EMAIL, PASSWORD, PASSWORD, 'Name required'),
+        ('A', LAST_NAME, EMAIL, PASSWORD, PASSWORD,
+         'Name has to be from 2 to 20 characters long'),
+        (NAME, '', EMAIL, PASSWORD, PASSWORD, 'Last name required'),
+        (NAME, 'B', EMAIL, PASSWORD, PASSWORD,
+         'Last name has to be from 2 to 20 characters long'),
+        (NAME, LAST_NAME, '', PASSWORD, PASSWORD, 'Email required'),
+        (NAME, LAST_NAME, 'Test123', PASSWORD, PASSWORD, 'Email is incorrect'),
+        (NAME, LAST_NAME, EMAIL, '', PASSWORD, 'Password required'),
+        (NAME, LAST_NAME, EMAIL, '123', PASSWORD,
+         'Password has to be from 8 to 15 characters long and contain '
+         'at least one integer, one capital, and one small letter'),
+        (NAME, LAST_NAME, EMAIL, PASSWORD, '', 'Re-enter password required'),
+        (NAME, LAST_NAME, EMAIL, BAD_PASSWORD, BAD_PASSWORD,
+         'Password has to be from 8 to 15 characters long and contain at '
+         'least one integer, one capital, and one small letter'),
+        (NAME, LAST_NAME, EMAIL, PASSWORD, MISMATCH_PASSWORD,
+         'Passwords do not match'),
+        (NAME, LAST_NAME, EMAIL, PASSWORD, PASSWORD, 'Registration complete'),
+    ],
+)
+def test_registration_through_sign_in_window(driver,
+                                             name, last_name, email,
+                                             password, re_password,
+                                             expected_message, test_logger):
+    """Tests registration."""
+    sign_up_page = SignUp(driver)
+    sign_in_page = SignIn(driver)
```

Comment:
In your test i saw that you reload page each time maybe will much faster do do all sign_up negative scenarios on already opened page  . Try this approach and please separate positive scenario from negative scenarios . Best practice -> do not mix positive and negative flows